lib/Controller/Stats.php

```diff
@@ -263,128 +274,56 @@ public function grid()
-        if (count($display_ids) <= 0)
-            throw new InvalidArgumentException(__('No displays with View permissions'), 'displays');
-
-        // Media on Layouts Ran
-        $select = '
-          SELECT stat.type,
-              display.Display,
-              layout.Layout,
-              IFNULL(`media`.name, IFNULL(`widgetoption`.value, `widget`.type)) AS Media,
-              COUNT(StatID) AS NumberPlays,
-              SUM(TIME_TO_SEC(TIMEDIFF(end, start))) AS Duration,
-              MIN(start) AS MinStart,
-              MAX(end) AS MaxEnd,
-              layout.layoutId,
-              stat.mediaId,
-              stat.widgetId,
-              stat.displayId
-        ';
-
-        $body = '
-            FROM stat
-              INNER JOIN display
-              ON stat.DisplayID = display.DisplayID
-              INNER JOIN layout
-              ON layout.LayoutID = stat.LayoutID
-              LEFT OUTER JOIN `widget`
-              ON `widget`.widgetId = stat.widgetId
-              LEFT OUTER JOIN `widgetoption`
-              ON `widgetoption`.widgetId = `widget`.widgetId
-                AND `widgetoption`.type = \'attrib\'
-                AND `widgetoption`.option = \'name\'
-              LEFT OUTER JOIN `media`
-              ON `media`.mediaId = `stat`.mediaId
-           WHERE stat.type <> \'displaydown\'
-                AND stat.end > :fromDt
-                AND stat.start <= :toDt
-                AND stat.displayID IN (' . implode(',', $display_ids) . ')
-        ';
-
-        $params = [
-            'fromDt' => $this->getDate()->getLocalDate($fromDt),
-            'toDt' => $this->getDate()->getLocalDate($toDt)
-        ];
-
-        // Type filter
-        if ($type == 'layout') {
-            $body .= ' AND `stat`.type = \'layout\' ';
-        } else if ($type == 'media') {
-            $body .= ' AND `stat`.type = \'media\' AND IFNULL(`media`.mediaId, 0) <> 0 ';
-        } else if ($type == 'widget') {
-            $body .= ' AND `stat`.type = \'media\' AND IFNULL(`widget`.widgetId, 0) <> 0 ';
-        }
-
-        // Layout Filter
-        if (count($layoutIds) != 0) {
-
-            $layoutSql = '';
-            $i = 0;
-            foreach ($layoutIds as $layoutId) {
-                $i++;
-                $layoutSql .= ':layoutId_' . $i . ',';
-                $params['layoutId_' . $i] = $layoutId;
-            }
-
-            $body .= '  AND `stat`.layoutId IN (' . trim($layoutSql, ',') . ')';
-        }
-
-        // Media Filter
-        if (count($mediaIds) != 0) {
-
-            $mediaSql = '';
-            $i = 0;
-            foreach ($mediaIds as $mediaId) {
-                $i++;
-                $mediaSql .= ':mediaId_' . $i . ',';
-                $params['mediaId_' . $i] = $mediaId;
+        /*Clear the display_ids array and add the user selected display in the array*/
+        if($displayId != null){
+            $display_ids = array();
+            if(!in_array($displayId, $display_ids)){
```

Comment:
As we refactor old code we make a habit of changing the names of variables where they don't fit - `$display_ids` could be refactored to `$displayIds`

Reply:
So I am not sure if this logic is right...

Take the following example... there are 5 displays, id 1,2,3,4,5
The user has permissions for id 1,2,3 but not 4,5
The user filters on id 1: They should only get stats for id 1.
The user then filters on id 4: They should get 0 records.
The user then filters on id 1 and 4: They should only get stats for id 1

I think the above code always gives them stats for 1,2,3 ?